doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+(serve-autoscaling)=
+
+# Autoscaling
+
+## Scaling horizontally with `num_replicas`
+
+One way to scale out your deployment without using autoscaling is by increasing the number of replicas. Each deployment consists of one or more [replicas](serve-architecture-high-level-view).
+The number of replicas is specified by the `num_replicas` field in the deployment options.
+By default, `num_replicas` is 1.
+
+```{literalinclude} doc_code/managing_deployments.py
+:start-after: __scaling_out_start__
+:end-before: __scaling_out_end__
+:language: python
+```
+
+## Autoscaling Important Concepts
+
+Instead of setting a fixed number of replicas for a deployment, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler adjusts to traffic spikes by observing queue sizes and making scaling decisions to add or remove replicas. To configure it, you can set the `autoscaling_config` field in deployment options.
```

Comment:
```suggestion
Instead of setting a fixed number of replicas for a deployment, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler reacts to traffic spikes by monitoring queue sizes and making scaling decisions to add or remove replicas. Configure it by setting the `autoscaling_config` field in deployment options.
```